velox/connectors/hive/storage_adapters/gcs/GcsFileSystem.cpp

```diff
@@ -36,12 +36,40 @@ using namespace connector::hive;
 namespace gcs = ::google::cloud::storage;
 namespace gc = ::google::cloud;
 
+namespace {
+
 auto constexpr kGcsInvalidPath = "File {} is not a valid gcs file";
 
+folly::Synchronized<
```

Comment:
Could we move credentialsProviderFactories and getCredentialsProviderByName method into the GcsOAuthCredentialsProvider class?

Reply:
`GcsOAuthCredentialsProvider` only defines the credentials provider interface. Moved them into `RegisterGcsFileSystem.cpp` to simplify the code.